cpp/oneapi/dal/table/backend/homogen_table_builder_impl.hpp

```diff
@@ -35,8 +37,8 @@ class homogen_table_builder_impl {
         dtype_ = data_type::float32;
     }
 
-    void reset(homogen_table&& t) {
-        if (t.has_data()) {
+    void move(detail::homogen_table_iface& t) override {
+        if (t.get_row_count() > 0 || t.get_column_count() > 0) {
```

Comment:
```suggestion
        if (t.get_row_count() > 0 && t.get_column_count() > 0) {
```
maybe you meant this?